monai/data/transforms/adaptors.py

```diff
@@ -0,0 +1,207 @@
+only necessary to name the dictionary keyword to which that value is assigned.
+
+Use of `outputs`
+
+`outputs` can take either a string, a list/tuple of string or a dict of string
+to string, depending on what the transform being adapted returns:
+. If the transform returns a single argument, then outputs can be supplied a
+  string that indicates what key to assign the return value to in the
+  dictionary
+. If the transform returns a list/tuple of values, then outputs can be supplied
+  a list/tuple of the same length. The strings in outputs map the return value
+  at the corresponding position to a key in the dictionary
+. If the transform returns a dictionary of values, then outputs must be supplied
+  a dictionary that maps keys in the function's return dictionary to the
+  dictionary being passed between functions
+
+Note, the caller is free to use a more complex way of specifying the outputs
+parameter than is required. The following are synonymous and will be treated
+identically:
+```
+   # single argument
+   adaptor(MyTransform(), 'image')
+   adaptor(MyTransform(), ['image'])
+   adaptor(MyTransform(), {'image': 'image'})
+
+   # multiple arguments
+   adaptor(MyTransform(), ['image', 'label'])
+   adaptor(MyTransform(), {'image': 'image', 'label': 'label'})
+```
+
+Use of `inputs`
+
+`inputs` can usually be omitted when using `adaptor`. It is only required when a
+the function's parameter names do not match the names in the dictionary that is
+used to chain transform calls.
+
+```
+class MyTransform1:
+    ...
+    def __call__(image):
+        return '''do stuff to image'''
+
+class MyTransform2:
+    ...
+    def __call__(img):
+        return '''do stuff to image'''
+
+d = {'image': i}
+
+Compose([
+    adaptor(MyTransform1(), 'image'),
+    adaptor(MyTransform2(), 'image', {'img':'image'})
+])
+```
+
+Inputs:
+dictionary in: None | Name maps
+params in (match): None | Name list | Name maps
+params in (mismatch): Name maps
+params & **kwargs (match) : None | Name maps
+params & **kwargs (mismatch) : Name maps
+
+Outputs:
+dictionary out: None | Name maps
+list/tuple out: list/tuple
+variable out: string
+
+"""
+@monai.utils.export('monai.data.transforms')
+def adaptor(function, outputs, inputs=None):
+
+    def must_be_types_or_none(variable_name, variable, types):
+        if variable is not None:
+            if not isinstance(variable, types):
+                raise ValueError(
+                    "'{}' must be None or {} but is {}".format(
+                        variable_name, types, type(variable)))
+
+    def must_be_types(variable_name, variable, types):
+        if not isinstance(variable, types):
+            raise ValueError(
+                "'{}' must be one of {} but is {}".format(
+                    variable_name, types, type(variable)))
+
+    def map_names(ditems, input_map):
+        return {k if k not in input_map else input_map[k]: v for k, v in ditems.items()}
```

Comment:
`k if k not in input_map else input_map[k]` -> `input_map.get(k,k)`

Reply:
I thought about it but came to the conclusion that input_map.get(k, k) would be less clear in its intent. I'm open to reevaluating.

Reply:
As we had discussed I think it's reasonable to expect people to know what `dict.get` does and that `get(k,k)` is understandable. The idiom `{remap.get(k,k):v for k,v in orig}` is understood to be a remapping of the keys in `orig`, where `remap` is a partial function from old keys to new. I'd say `if-else` expressions is more verbose and harder to read, and may be less obvious that it binds tighter than `:`.